src/backends/multicore/threshold_watcher.hpp

```diff
@@ -52,37 +59,34 @@ class threshold_watcher {
     /// Reset state machine for each detector.
     /// Assume that the values in values_ have been set correctly before
     /// calling, because the values are used to determine the initial state
-    void reset(value_type t=0) {
+    void reset() {
         clear_crossings();
         for (auto i=0u; i<size(); ++i) {
-            is_crossed_[i] = values_[index_[i]]>=thresholds_[i];
+            is_crossed_[i] = values_[cv_index_[i]]>=thresholds_[i];
         }
-        t_prev_ = t;
     }
 
     const std::vector<threshold_crossing>& crossings() const {
         return crossings_;
     }
 
-    /// The time at which the last test was performed
-    value_type last_test_time() const {
-        return t_prev_;
-    }
-
     /// Tests each target for changed threshold state
     /// Crossing events are recorded for each threshold that
     /// is crossed since the last call to test
-    void test(value_type t) {
+    void test() {
         for (auto i=0u; i<size(); ++i) {
+            auto cv     = cv_index_[i];
```

Comment:
We could, but given the shortness of the method, it seems to me to be an unnecessary qualification.